donkeycar/templates/basic.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+"""
+Scripts to drive a donkey car
+
+Usage:
+    drive.py [--model=<model>] [--type=(linear|categorical|tflite_linear)]
+
+Options:
+    -h --help          Show this screen.
+"""
+
+from docopt import docopt
+
+import donkeycar as dk
+from donkeycar.parts.tub_v2 import TubWriter
+from donkeycar.parts.controller import LocalWebController
+from donkeycar.parts.camera import PiCamera
+from donkeycar.parts.image import ImgPrecondition
+from donkeycar.parts.actuator import PCA9685, PWMSteering, PWMThrottle
+
+
+class DriveMode:
+    """ Helper class to dispatch between ai and user driving"""
+
+    def __init__(self, cfg):
+        self.cfg = cfg
+
+    def run(self, mode, user_angle, user_throttle, pilot_angle, pilot_throttle):
+        if mode == 'user':
+            return user_angle, user_throttle
+        elif mode == 'local_angle':
+            return pilot_angle if pilot_angle else 0.0, user_throttle
+        else:
+            return pilot_angle if pilot_angle else 0.0, \
+                   pilot_throttle * self.cfg.AI_THROTTLE_MULT if \
+                       pilot_throttle else 0.0
+
+
+class PilotCondition:
+    """ Helper class to determine how is in charge of driving"""
+    def run(self, mode):
+        return mode != 'user'
+
+
+def drive(cfg, model_path=None, model_type=None):
+    """
+    Construct a minimal robotic vehicle from many parts. Here, we use a
+    single camera, web or joystick controller, autopilot and tubwriter.
+
+    Each part runs as a job in the Vehicle loop, calling either it's run or
+    run_threaded method depending on the constructor flag `threaded`. All
+    parts are updated one after another at the framerate given in
+    cfg.DRIVE_LOOP_HZ assuming each part finishes processing in a timely
+    manner. Parts may have named outputs and inputs. The framework handles
+    passing named outputs to parts requesting the same named input.
+    """
+
+    car = dk.vehicle.Vehicle()
+    # add camera
+    cam = PiCamera(image_w=cfg.IMAGE_W, image_h=cfg.IMAGE_H,
```

Comment:
Should we checking for `cfg.camera` to see which one to use?

Reply:
Yes, let me add that.